SuperHero/src/test/java/ApiTests.java

```diff
@@ -0,0 +1,280 @@
+		assertEquals("fight", actualMainSkill, "Основной навык не соответствует ожидаемому");
+	}
+
+	@Test
+	@DisplayName("Validate required fields")
+	void postTest2() {
+		String body = """
+				    {
+				        "birthDate": "",
+				        "city": "",
+				        "fullName": "",
+				        "gender": "",
+				        "mainSkill": ""
+				    }
+				""";
+		ValidatableResponse response =
+				given().
+						header("accept", "application/json").
+						header("Content-Type", "application/json").
+						body(body).
+						when().
+						post(baseUrl).then();
+		Response response1 = response.extract().response();
+		int status = response1.statusCode();
+		//баг: неверный статус код: 403 вместо 400
+		Assertions.assertEquals(400, status, "Статус код не соответствует ожидаемому");
+		String expectedMessage = "{\n" +
+				"    \"message\": \"Incorrect request data\",\n" +
+				"    \"code\": \"BAD_REQUEST\"\n" +
+				"}";
+		String actualMessage = response1.body().asString();
+		Assertions.assertEquals(expectedMessage, actualMessage, "Сообщение в теле ответа не соответствует ожиданиям");
+	}
+
+
+	@Test
+	@DisplayName("GET all superheroes")
+	void getTest1() {
+		given().
+				when().
+				get(baseUrl).
+				then().
+				log().
+				all().
+				statusCode(200).
+				assertThat()
+				.body(notNullValue()) // Проверка, что тело ответа не null
+				.body(not(isEmptyOrNullString())); // Проверка, что тело ответа не пустое
+	}
+
+	@Test
+	@DisplayName("GET superhero by Id")
+	void getTest2() {
+		int id = 3;
+		given().
+				when().
+				get(baseUrl + "/" + id)
+				.then().
+				log().
+				all().
+				statusCode(200).
+				body("id", equalTo(id));
+	}
+
+	@Test
+	@DisplayName("GET superhero by not existed id")
+	void getTest3() {
+		int id = 154;
+		given().
+				when().
+				get(baseUrl + "/" + id)
+				.then().
+				log().
+				all().
+				//баг: неверный статус код: 400 вместо 404
+						statusCode(404).
+				body("id", not(equalTo(id)));
+	}
+
+	@Test
+	@DisplayName("update superhero all fields")
+	void putTest1() {
+		int id = 1;
+		String body = """
+				    {
+				        "birthDate": "2020-02-21",
+				        "city": "PARIS",
+				        "fullName": "LADY BUG",
+				        "gender": "F",
+				        "mainSkill": "BUGS"
+				    }
+				""";
+
+		ValidatableResponse response =
+				given()
+						.header("accept", "application/json")
+						.header("Content-Type", "application/json")
+						.body(body)
+						.when()
+						.put(baseUrl + "/" + id)
+						.then()
+						.body("id", equalTo(id))
+						.statusCode(200);
+
+
+		response.log().body();
+		given()
+				.header("accept", "application/json")
+				.when()
+				.get(baseUrl + "/" + id)
+				.then()
+				.log().all()
+				.statusCode(200)
+				.body("city", equalTo("PARIS"))
+				.body("fullName", equalTo("LADY BUG"))
+				.body("birthDate", equalTo("2020-02-21"))
+				.body("gender", equalTo("F"))
+				.body("mainSkill", equalTo("BUGS"));
```

Comment:
Actually good alternative way to check the data without JUnit assertions, but issue here is when the init data will be changed you will need to fix the scenario explicitly